File: app/src/main/java/io/github/shadow578/music_dl/util/storage/StorageKey.java
package io.github.shadow578.music_dl.util.storage;

import androidx.annotation.NonNull;

import io.github.shadow578.music_dl.KtPorted;

/**
 * a storage key, used by {@link StorageHelper}
 */
@KtPorted
public class StorageKey {

    /**
     * a empty storage key
     */
    public static final StorageKey EMPTY = new StorageKey("");

    /**
     * the internal string key
     */
    private final String key;

    /**
     * create a new storage key
     *
     * @param key the key of the file
     */
    public StorageKey(String key) {
        this.key = key;
    }

    /**
     * get the key as a string
     *
     * @return the string key
     */
    @NonNull
    @Override
    public String toString() {
        return key;
    }
}

File: app/src/main/kotlin/io/github/shadow578/yodel/util/storage/StorageKey.kt
package io.github.shadow578.yodel.util.storage

/**
 * a storage key, used by util/StorageHelper.kt
 */
data class StorageKey(
    val key: String
) {

    override fun toString(): String {
        return key
    }

    companion object {
        /**
         * a empty storage key
         */
        val EMPTY = StorageKey("")
    }
}